tests/ResultQueryTest.php

```diff
@@ -588,4 +588,164 @@ public function testQueryStreamFromLazyConnectionWillErrorWhenConnectionIsClosed
+                if ($current < $min) {
+                    throw new \Exception('max_allowed_packet too low: current: ' . $current . ' min: ' . $min);
+                }
+                return \React\Promise\resolve();
+            }
+        )->then(
+            function () {
+                return true;
+            }
+        );
+    }
+
+    /**
+     * This should not trigger splitted packets sending
+     */
+    public function testSelectStaticTextSplitPacketsExactlyBelow16MiB()
+    {
+        $connection = $this->createConnection(Loop::get());
+
+        $promise = $this->checkMaxAllowedPacket($connection, 0x1000000); // 16MiB
+
+        $promise->then(
+            function () use ($connection) {
+                /**
+                 * This should be exactly below 16MiB packet
+                 *
+                 * x03 + "select ''" = len(10)
+                 */
+                $text = str_repeat('A', 0xffffff - 11);
+                $connection->query('select \'' . $text . '\'')->then(function (QueryResult $command) use ($text) {
+                    $this->assertCount(1, $command->resultRows);
+                    $this->assertCount(1, $command->resultRows[0]);
+                    $this->assertSame($text, reset($command->resultRows[0]));
+                })->done();
+            }
+        )->otherwise(
+            function (\Throwable $e) {
+                $this->markTestIncomplete('checkMaxAllowedPacket: ' . $e->getMessage());
+            }
+        )->always(
+            function () use ($connection) {
+                $connection->quit();
+            }
+        )->done();
+
+        Loop::run();
+    }
+
+    /**
+     * This should trigger split packets sending and
+     * will send additional empty packet to signal to the server that split packets has ended.
+     */
+    public function testSelectStaticTextSplitPacketsExactly16MiB()
+    {
+        $connection = $this->createConnection(Loop::get());
+
+        $promise = $this->checkMaxAllowedPacket($connection);
+
+        $promise->then(
+            function () use ($connection) {
+                /**
+                 * This should be exactly at 16MiB packet
+                 *
+                 * x03 + "select ''" = len(10)
+                 */
+                $text = str_repeat('A', 0xffffff - 10);
+                $connection->query('select \'' . $text . '\'')->then(function (QueryResult $command) use ($text) {
+                    $this->assertCount(1, $command->resultRows);
+                    $this->assertCount(1, $command->resultRows[0]);
+                    $this->assertSame($text, reset($command->resultRows[0]));
+                })->done();
+            }
+        )->otherwise(
+            function (\Throwable $e) {
+                $this->markTestIncomplete('checkMaxAllowedPacket: ' . $e->getMessage());
+            }
+        )->always(
+            function () use ($connection) {
+                $connection->quit();
```

Comment:
The [`PromiseInterface::always()`](https://github.com/reactphp/promise#promiseinterfacealways) and [`PromiseInterface::otherwise()`](https://github.com/reactphp/promise#promiseinterfaceotherwise) functions are marked as deprecated as described in [reactphp/promise](https://github.com/reactphp/promise), I don't think it's a good idea to use them in this context as we want to avoid using deprecated functions.

Reply:
This only works on react-promise:3.x ... not react-promise:2.7.

Will only react-promise:3.x be supported? In current package.json both are supported.

Reply:
You're right that both versions are supported and we already know that v3 will be the way forward. I agree that it should be compatible with both versions at this point, but looking into the future here. If we already know that this functionality will be removed at a later point in time, it makes sense to avoid using these functions, so we don't have to remove them again.

Reply:
Yeah, but 3.x introduced that, 2.7 does not work with `catch` and `finally`. So, when 2.7 version is removed from this package, it will be time to also change those methods.